controller/src/main/java/io/pravega/controller/task/Stream/StreamTransactionMetadataTasks.java

```diff
@@ -700,6 +703,30 @@ private PingTxnStatus getPingTxnStatus(final TxnStatus txnStatus) {
         }, executor);
     }
 
+    /**
+     * Method to garbage collect (abort) open transactions that have lived past their lease period.
+     * @param scope scope
+     * @param stream stream
+     * @param openTransactionData Metadata for open transactions on the Stream
+     * @param contextOpt operation context
+     * @return future.
+     */
+    public CompletableFuture<Void> transactionGC(final String scope, final String stream, final Map<UUID, ActiveTxnRecord> openTransactionData,
+                                                 final OperationContext contextOpt) {
+        Preconditions.checkNotNull(contextOpt);
+        Preconditions.checkNotNull(openTransactionData);
+        List<CompletableFuture<Void>> txnFuturesList = openTransactionData.entrySet().stream().map(txnEntry -> {
```

Comment:
In the PR's description it says that the open time can be set to max long. So new transactions can be created but never time out. Yet here it loads all open transactions in memory and iterates over them. This function's time an memory requirements will grow linearly depending on how long the system is left running and eventually OOM.

Reply:
@tkaitchuck Yes, Setting the txn expiry time to max long will essentially mean the txn never timesout. 


The PR at this point does not have all the code changes & is still in draft mode and not ready for review.

Reply:
This is not a question of rate limiting. The system should function with billions of open transactions. If it does not, it is a bug.
If this PR is not addressing this bug, that's one thing, but it appears to be actively be making it worse when the stated purpose is to help handle larger numbers of transactions.